include/openssl/ec_key.h

```diff
@@ -356,6 +328,115 @@ OPENSSL_EXPORT EC_KEY *o2i_ECPublicKey(EC_KEY **out_key, const uint8_t **inp,
+// should not free this struct.
+//
+// This struct is also zero-initialized. This is different from OpenSSL which
+// returns function pointers to the default implementations within the
+// |EC_KEY_METHOD| struct. We do not do this to make it easier for the
+// compiler/linker to drop unused functions. The wrapper functions will select
+// the appropriate |ec_key_default_*| implementation.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_OpenSSL(void);
+
+// EC_KEY_METHOD_new returns a newly allocated |EC_KEY_METHOD| object. If the
+// input parameter |eckey_meth| is non-NULL, the function pointers within the
+// returned |EC_KEY_METHOD| object will be initialized to the values from
+// |eckey_meth|. If |eckey_meth| is NULL, the returned object will be
+// initialized using the value returned from |EC_KEY_get_default_method|.
+OPENSSL_EXPORT EC_KEY_METHOD *EC_KEY_METHOD_new(const EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_METHOD_free frees the memory associated with |eckey_meth|
+OPENSSL_EXPORT void EC_KEY_METHOD_free(EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_set_method sets |meth| on |ec|. While we provide setters to set
+// the |copy|, |set_group|, |set_private|, |set_public|, and |sign_setup|
+// fields, we do not support the associated functionality and these pointers
+// should be set to NULL. We do not support the |verify|, |verify_sig|, or
+// |keygen| fields yet.
+//
+// Returns zero on failure and one on success.
+OPENSSL_EXPORT int EC_KEY_set_method(EC_KEY *ec, const EC_KEY_METHOD *meth);
+
+// EC_KEY_get_method returns the |EC_KEY_METHOD| object associated with |ec|.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_get_method(const EC_KEY *ec);
+
+#ifdef __cplusplus
+#define IS_NULL(ptr) ((ptr) == nullptr)
+#else
+#define IS_NULL(ptr) ((ptr) == NULL)
+#endif
+
+// EC_KEY_METHOD_set_sign_impl sets function pointers on |meth|.
+// AWS-LC currently supports setting |sign| and |sign_sig|. |sign_setup| must
+// be set to NULL in order to compile with AWS-LC. At runtime, if |sign_setup|
+// is not a NULL field, the function aborts.
+OPENSSL_EXPORT void EC_KEY_METHOD_set_sign_impl(EC_KEY_METHOD *meth,
+                                int (*sign)(int type, const uint8_t *digest,
+                                        int digest_len, uint8_t *sig,
+                                        unsigned int *siglen,
+                                        const BIGNUM *k_inv,
+                                        const BIGNUM *r, EC_KEY *eckey),
+                                int (*sign_setup)(EC_KEY *eckey, BN_CTX *ctx_in,
+                                        BIGNUM **kinvp, BIGNUM **rp),
+                                ECDSA_SIG *(*sign_sig)(const uint8_t *digest,
+                                         int digest_len,
+                                         const BIGNUM *in_kinv,
+                                         const BIGNUM *in_r,
+                                         EC_KEY *eckey));
```

Comment:
* Since `EC_KEY_METHOD_set_sign_impl` is our own function, there's no need for it to accept the parameters (like `sign_setup`) that it doesn't support.
* Is `EC_KEY_METHOD_set_sign_impl` a function we want public?  Is the `_impl` suffix a pattern that we want to establish as a precedent if we face this type of issue in the future? Would a `_awslc` suffix be better?

Reply:
Removed the parameters and changed the suffix to _awslc. I don't think it matters if our variant is public or not, it should be okay for consumers to use either function.